src/platform_impl/linux/x11/ime/inner.rs

```diff
@@ -5,7 +5,9 @@ use super::{ffi, XConnection, XError};
 use super::{context::ImeContext, input_method::PotentialInputMethods};
 
 pub unsafe fn close_im(xconn: &Arc<XConnection>, im: ffi::XIM) -> Result<(), XError> {
-    (xconn.xlib.XCloseIM)(im);
+    if !im.is_null() {
```

Comment:
Huh, I swear someone had already done this.

Anyways, we should issue a warning if it is null.